### PR TITLE
LUCENE-9391: Upgrade HPPC to 0.8.2

### PR DESCRIPTION
**Pull request by zhaih**
# Description

Upgrade HPPC to 0.8.2
`ant test` passed
`ant precommit` passed

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `master` branch.
- [x] I have run `ant precommit` and the appropriate test suite.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

**Comment by dweiss** [comment]
Hi Patrick. Would you add a corresponding CHANGES entry (to both Solr and Lucene, I guess) and attribution to your name? Otherwise LGTM.

**Comment by zhaih** [comment]
> Hi Patrick. Would you add a corresponding CHANGES entry (to both Solr and Lucene, I guess) and attribution to your name? Otherwise LGTM.

Thanks for reminding me that! Added!

File: lucene/ivy-versions.properties [hunk 1]
```diff
@@ -10,7 +10,7 @@ com.carrotsearch.randomizedtesting.version = 2.7.6
 
 /com.carrotsearch.thirdparty/simple-xml-safe = 2.7.1
 
-/com.carrotsearch/hppc = 0.8.1
+/com.carrotsearch/hppc = 0.8.2
```
[the remark this answers is not in the document]
Comment (zhaih) [reply lucene/ivy-versions.properties:13]: I think so? In `facet`'s `ivy.xml` it use `rev="${/com.carrotsearch/hppc}"` which I believe is the variable here.

**Comment by danmuzi** [comment]
Hi Patrick,
Thanks for your contribution 👍 
I found the JIRA-issue number of this PR is wrong.
It should be changed from [LUCENE-8574] to [LUCENE-9391].
https://issues.apache.org/jira/browse/LUCENE-9391
Please check it.

**Comment by zhaih** [comment]
> Hi Patrick,
> Thanks for your contribution 👍
> I found the JIRA-issue number of this PR is wrong.
> It should be changed from [LUCENE-8574] to [LUCENE-9391].
> https://issues.apache.org/jira/browse/LUCENE-9391
> Please check it.

Oh, thank you for figuring that out! Yeah I picked up a wrong one from my backlog... Thank you very much!

**Comment by zhaih** [comment]
Do I need to squash the commits? Seems commits in Lucene are all squashed? Or it will be done automatically when merging somehow?

**Comment by danmuzi** [comment]
Please do the **"Squash and merge"** below.
Your sub commits will be combined automatically.

**Pull request merged**